physics/satmedmfvdifq.F

```diff
@@ -1117,8 +1193,14 @@ subroutine satmedmfvdifq_run(im,km,progsigma,ntrac,ntcw,ntrw,     &
           else
             ptem = 1. + 2.7 * zol(i)
             zk = tem / ptem
-          endif 
-          elm(i,k) = zk*rlam(i,k)/(rlam(i,k)+zk)
+          endif
+
+          if (tc_pbl == 0) then
+            elm(i,k) = zk*rlam(i,k)/(rlam(i,k)+zk)
+          else if (tc_pbl == 1) then
+            ! new blending method for mixing length
+            elm(i,k) = sqrt( 1.0/( 1.0/(zk**2)+1.0/(rlam(i,k)**2) ) )
+          endif
 
 !>  - If sfc_rlm=1, use zk for elm within surface layer
```

Comment:
Can we move the line 1205-1209 after line 1199? This change is related to the configuration of HAFS-A (under tc_pbl=0) and do not need to be included for HAFS-B since our changes already have this function.

Reply:
@chxmoonlight, done!